### ISSUE TITLE
One-path spec from all-path suite that fails

### ISSUE THREAD

**Issue by ana-pantilie** [kframework/kore#1289]
The following `one-path` spec should pass. Right now only https://github.com/kframework/kore/blob/master/src/main/k/working/all-path/03-unification/set-select/provable/one-path-b-or-c-spec.k passes.

```
module VERIFICATION
  imports PATH

endmodule

module ONE-PATH-B-OR-C-SPEC
  imports VERIFICATION
  imports PATH

  rule <k> select => b #Or c </k>
         <state> SetItem(a) SetItem(b) SetItem(c) </state> [one-path]

endmodule
```

**Comment by ttuegel** [comment]
We need to investigate this bug and see if it is still valid.

**Comment by ana-pantilie** [comment]
@traiansf and I investigated this and the conclusion we reached is that this spec fails because the current implementation for one-path doesn't consider non-deterministic semantics. We couldn't find a straightforward way to modify the current one-path strategy to account for such behavior. 

It also looks like the following integration test https://github.com/kframework/kore/blob/master/test/all-path/03-unification/set-select/one-path-b-spec.k falls in the same category as the one from this issue but passes by accident.
